test/test-browser-usage.js

```diff
@@ -0,0 +1,212 @@
+      return globalDriver.get(localURL)
+      .then(() => {
+        return globalDriver.wait(selenium.until.titleIs('Example Site'), 1000);
+      });
+    })
+    .then(() => {
+      return seleniumAssistant.killWebDriver(globalDriver);
+    })
+    .then(() => {
+      globalDriver = null;
+    });
+  }
+
+  function testBuilderSeleniumUsage(specificBrowser) {
+    const builder = specificBrowser.getSeleniumDriverBuilder();
+
+    return builder.buildAsync()
+    .then(driver => {
+      globalDriver = driver;
+    })
+    .then(() => {
+      return globalDriver.get(localURL);
+    })
+    .then(() => {
+      return globalDriver.wait(selenium.until.titleIs('Example Site'), 1000);
+    })
+    .then(() => {
+      return seleniumAssistant.killWebDriver(globalDriver);
+    })
+    .then(() => {
+      globalDriver = null;
+    });
+  }
+
+  function testBrowserInfo(specificBrowser) {
+    const versionString = specificBrowser.getRawVersionString();
+    (versionString === null).should.equal(false);
+
+    const versionNumber = specificBrowser.getVersionNumber();
+    versionNumber.should.not.equal(-1);
+  }
+
+  browserIds.forEach(browserId => {
+    releases.forEach(release => {
+      if (browserId === 'safari' && release === 'unstable') {
+        // Safari unstable doesn't exist so skip it.
+        return;
+      }
+
+      if (browserId === 'safari') {
+        // Safari fails (both stable and beta) with
+        // "Error: Server terminated early with status 1".
+        return;
+      }
+
+      let specificBrowser = seleniumAssistant.getBrowser(browserId, release);
+      if (!specificBrowser) {
+        console.warn(chalk.red('WARNING:') + ' Unable to find ' +
+          browserId + ' ' + release);
+        return;
+      }
+
+      it(`should be able to use ${browserId} - ${release}`, function() {
+        this.timeout(DOWNLOAD_TIMEOUT);
+
+        if (!specificBrowser.isValid()) {
+          if (browserId === 'opera') {
+            console.warn('Opera not available in this environment.');
+            return;
+          }
+
+          throw new Error(`Browser is unexpectidly invalid ${browserId} - ${release}`);
+        }
+
+        let afterDownloadPath = specificBrowser.getExecutablePath();
+        if (browserId === 'opera') {
+          if (process.platform === 'darwin') {
+            afterDownloadPath.indexOf(
+              path.normalize('/Applications/Opera')
+            ).should.not.equal(-1);
+          } else {
+            console.warn('Unable to ensure location of Opera on this ' +
+              'platform.');
+          }
+        } else {
+          afterDownloadPath.indexOf(
+            path.normalize(seleniumAssistant.getBrowserInstallDir())
+          ).should.not.equal(-1);
+        }
+
+        if (isBlackListed(specificBrowser)) {
+          console.log(`Skipping ${browserId} - ${release} due to blacklist.`);
+          return;
+        }
+
+        return testNormalSeleniumUsage(specificBrowser)
```

Comment:
Here's a case where both single-line arrow functions and indenting your `.then()`s help 👓 :

```js
return testNormalSeleniumUsage(specificBrowser)
  .then(() => testBuilderSeleniumUsage(specificBrowser))
  .then(() => testBrowserInfo(specificBrowser));
```